CRM/SMS/BAO/Provider.php

```diff
@@ -92,6 +93,9 @@ public static function getProviders($selectArr = NULL, $filter = NULL, $getActiv
   public static function saveRecord($values) {
     $dao = new CRM_SMS_DAO_Provider();
     $dao->copyValues($values);
+    if (empty($dao->domain_id)) {
+      $dao->domain_id = CRM_Core_Config::domainID();
```

Comment:
Have now updated using the style found here https://github.com/civicrm/civicrm-core/blob/648631cd94799e87fe2347487d465b1a7256aa57/CRM/Core/BAO/StatusPreference.php#L70